### PR TITLE
feat(cxx): add --emit_anchor_scopes and implement childof edges for lambdas

### PR DESCRIPTION
**Pull request by justbuchanan**

### CODE REVIEW

File: kythe/cxx/indexer/cxx/KytheGraphObserver.cc [hunk 1]
```diff
@@ -756,6 +756,14 @@ void KytheGraphObserver::recordChildOfEdge(const NodeId& child_id,
                      VNameRefFromNodeId(parent_id));
 }
 
+void KytheGraphObserver::recordChildOfEdge(const Range& source_range,
+                                           const NodeId& parent_id) {
+  RecordAnchor(source_range, parent_id, EdgeKindID::kChildOf,
+               Claimability::Claimable);
+  // TODO(justbuchanan): stamped?
+  // TODO(justbuchanan): claimable?
```
Comment (justbuchanan) [inline kythe/cxx/indexer/cxx/KytheGraphObserver.cc:764]: I'm not clear on when things should be stamped and/or claimable.

Comment (shahms) [reply kythe/cxx/indexer/cxx/KytheGraphObserver.cc:764]: stamped is for template instantiations, IIRC. Can't remember on Claimable, though.

File: kythe/cxx/indexer/cxx/IndexerASTHooks.cc [hunk 2]
```diff
@@ -3488,6 +3492,22 @@ bool IndexerASTVisitor::VisitFunctionDecl(clang::FunctionDecl* Decl) {
...
+    if (absl::GetFlag(FLAGS_emit_anchor_scopes) && lambda &&
+        DefinitionRangeInContext.has_value()) {
+      if (Job->BlameStack.size() < 2) {
+        // TODO(justbuchanan): add childof edge from the lambda to the file node
+      } else {
+        // The last entry in the blame stack is the lambda we're currently
+        // looking at. We need to go one up from that to find the enclosing
+        // function
+        for (const auto& Caller : Job->BlameStack.rbegin()[1]) {
+          Observer.recordChildOfEdge(DefinitionRangeInContext.value(), Caller);
```
Comment (shahms) [inline kythe/cxx/indexer/cxx/IndexerASTHooks.cc:3507]: Just `*DefinitionRangeInContext` after checking for validity (as is done above).

File: kythe/cxx/indexer/cxx/KytheGraphObserver.cc (hunk 1 above)
Comment (justbuchanan) [reply kythe/cxx/indexer/cxx/KytheGraphObserver.cc]: `Claimable` seems to match similar invocations, so I'll leave this as-is.

File: kythe/cxx/indexer/cxx/IndexerASTHooks.cc (hunk 2 above)
Comment (justbuchanan) [reply kythe/cxx/indexer/cxx/IndexerASTHooks.cc]: Done

**Comment by shahms** [comment]
seems like an unrelated build failure?

**Comment by justbuchanan** [comment]
> seems like an unrelated build failure?

Yeah looks like a verifier test timeout.

**Review by shahms — APPROVED** [review]

**Pull request merged**